cli/src/semgrep/target_manager.py

```diff
@@ -532,7 +535,7 @@ class TargetManager:
     respect_rule_paths: bool = True
     baseline_handler: Optional[BaselineHandler] = None
     allow_unknown_extensions: bool = False
-    file_ignore: Optional[FileIgnore] = None
+    ignore_profiles: Dict[str, FileIgnore] = {}
```

Comment:
Does this need to be a factory? I think this may result in this dictionary getting shared by everything (desired?).